validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -90,29 +92,47 @@ public void validate() throws Exception {
     RetryHelper.retry(
         maxRetryCount,
         () -> {
-          // We will query both the Service and RemoteService dimensions to ensure we get all
-          // metrics from all aggregations, specifically the [RemoteService] aggregation.
+          // We will query metrics that include Service, RemoteService, or RemoteTarget dimensions
+          // to ensure we get all metrics from all aggregations, specifically the [RemoteService] aggregation.
           List<String> serviceNames =
               Lists.newArrayList(
                   context.getServiceName(), context.getRemoteServiceDeploymentName());
-          // TODO - Put the following back in: "www.amazon.com",  "AWS.SDK.S3"
+          // TODO - Put the following back in: "www.amazon.com"
           List<String> remoteServiceNames =
               Lists.newArrayList(context.getRemoteServiceDeploymentName());
           if (context.getRemoteServiceName() != null && !context.getRemoteServiceName().isEmpty()) {
             serviceNames.add(context.getRemoteServiceName());
           }
 
           List<Metric> actualMetricList = Lists.newArrayList();
-          addMetrics(
-              CloudWatchService.SERVICE_DIMENSION,
-              serviceNames,
-              expectedMetricList,
-              actualMetricList);
-          addMetrics(
-              CloudWatchService.REMOTE_SERVICE_DIMENSION,
-              remoteServiceNames,
-              expectedMetricList,
-              actualMetricList);
+
+          // Add sets of dimension filters to use for each query to CloudWatch.
+          List<List<Pair<String, String>>> dimensionLists = Lists.newArrayList();
+          // Query metrics that includes any of these <service> values.
+          for (String serviceName : serviceNames) {
+            dimensionLists.add(
+                Arrays.asList(new Pair<>(CloudWatchService.SERVICE_DIMENSION, serviceName)));
+          }
+          // Query metrics that includes any of these <remoteService> values.
+          for (String remoteServiceName : remoteServiceNames) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, remoteServiceName)));
+          }
+          // Query for metrics that includes both of these <remoteService, remoteTarget> values.
+          // Querying just 'remoteService="AWS.SDK.S3"' would also work, but that can result in
+          // returning too many canary metrics and may cause issues.
+          if (context.getTestingId() != null && !context.getTestingId().isEmpty()) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, "AWS.SDK.S3"),
+                    new Pair<>(CloudWatchService.REMOTE_TARGET_DIMENSION, "::s3:::e2e-test-bucket-name-" + context.getTestingId())));
```

Comment:
Hardcoding `AWS.SDK.S3` and `::s3:::e2e-test-bucket-name-` is not great. This value should be pulled from the expected data templates if possible. I can see how that would be difficult though so this should be fine.

Reply:
Since values are usually provided to the templates, not pulled from them, another alternative to remove hard-coded values is to [undo this previous change](https://github.com/aws-observability/aws-application-signals-test-framework/pull/41/commits/d21a9e07ef76e3627396458cdee0dc4d796d6f14) and keep usage of `--remote-target-name`, and apply similar logic for `AWS.SDK.S3`. Unless we we want that logic back, I'll leave this as is.

Reply:
No, I think this is fine for now since we are looking for very specific values. Is looking for `RemoteService == AWS.SDK.S3` even necessary though, given we will also be looking for the `RemoteTarget` which is a unique value?